agent/agent_endpoint_test.go

```diff
@@ -391,17 +412,19 @@ func TestAgent_Services_ACLFilter(t *testing.T) {
 	})
 
 	t.Run("root token", func(t *testing.T) {
-		require := require.New(t)
-
-		req := httptest.NewRequest("GET", "/v1/agent/services?token=root", nil)
-		rsp := httptest.NewRecorder()
-
-		obj, err := a.srv.AgentServices(rsp, req)
-		require.NoError(err)
+		req, _ := http.NewRequest("GET", "/v1/agent/services?token=root", nil)
+		resp := httptest.NewRecorder()
+		a.srv.h.ServeHTTP(resp, req)
+		dec := json.NewDecoder(resp.Body)
+		var val map[string]*api.AgentService
+		err := dec.Decode(&val)
+		if err != nil {
+			t.Fatalf("Err: %v", err)
+		}
 
-		val := obj.(map[string]*api.AgentService)
-		require.Len(val, 2)
-		require.Empty(rsp.Header().Get("X-Consul-Results-Filtered-By-ACLs"))
+		if len(val) != 1 {
```

Comment:
The previous assertion was for len=2, not 1

Also, I'd prefer using package `require` here for readability